### PR TITLE
Deprecate SbDirectoryCreate

### PR DESCRIPTION
**Pull request by madhurajayaraman**
b/302730696

### CODE REVIEW

File: starboard/elf_loader/exported_symbols.cc [hunk 1]
```diff
@@ -47,6 +48,9 @@
 #if SB_API_VERSION >= 16
 #include "starboard/shared/modular/posix_mmap_wrappers.h"
 #include "starboard/shared/modular/posix_time_wrappers.h"
+#ifdef _WIN32
+#include <direct.h>
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/elf_loader/exported_symbols.cc:52]: Tried building on Windows without this, and got an error

File: starboard/linux/shared/system_get_path.cc [hunk 2]
```diff
@@ -46,7 +46,8 @@ bool GetCacheDirectory(char* out_path, int path_size) {
     out_path[0] = '\0';
     return false;
   }
-  return SbDirectoryCreate(out_path);
+  bool exists = SbDirectoryCanOpen(out_path);
+  return (mkdir(out_path, 0700) == 0 || exists);
```
[the remark this answers is not in the document]
Comment (madhurajayaraman) [reply starboard/linux/shared/system_get_path.cc:50]: All of these are about to change in the followup 'Deprecate SbDirectoryCanOpen" CL -- if there are other changes to this CL I need to make, I can swap order but I should have a followup soon where I'll fix this. I think there are a couple of other places I can flip these for efficiency.

**Comment by andrewsavage1** [comment]
Here we use both the windows api and the posix api where in general we would just use the posix api have windows mimic it as required, with later considerations to directly use the windows api. Why the change in this PR?

File: base/files/file_util_starboard.cc [hunk 3]
```diff
@@ -316,8 +321,15 @@ bool CreateDirectoryAndGetError(const FilePath &full_path, File::Error* error) {
   AssertBlockingAllowed();
 
   // Fast-path: can the full path be resolved from the full path?
+  int retval;
+  #ifdef _WIN32
+    retval = _mkdir(full_path.value().c_str());
```
Comment (y4vor) [inline base/files/file_util_starboard.cc:326]: We shouldn't be using the win32 APIs. 
Can you add a posix_emu thing that provide `mkdir`?

Comment (madhurajayaraman) [reply base/files/file_util_starboard.cc]: Fixed

File: starboard/shared/win32/posix_emu/stat.cc [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/stat.cc:1]: 2024

**Review by y4vor — COMMENTED** [review]
Please update starboard/CHANGELOG.md

File: starboard/CHANGELOG.md [hunk 5]
```diff
@@ -64,6 +64,10 @@ Config OBUs as side data.
 ### Deprecated SbStringScan and SbStringScanF
 The APIs defined in `starboard/string.h` are deprecated and the standard API `vsscanf` and `sscanf` are used instead.
 
+### Decrecated SbTime APIs and migrated to POSIX time APIs
```
Comment (y4vor) [inline starboard/CHANGELOG.md:67]: Change SbTime to SbDirectoryCreate.

Comment (madhurajayaraman) [reply starboard/CHANGELOG.md]: Oops

**Review by y4vor — APPROVED** [review]

File: base/files/file_util_starboard.cc [hunk 6]
```diff
@@ -231,7 +232,7 @@ bool CopyFile(const FilePath &from_path, const FilePath &to_path) {
 
 bool PathExists(const FilePath &path) {
   AssertBlockingAllowed();
-  return SbFileExists(path.value().c_str());
+    return SbFileExists(path.value().c_str());
```
Comment (andrewsavage1) [inline base/files/file_util_starboard.cc:235]: This spacing looks incorrect, make sure to clang-format this file

**Review by andrewsavage1 — APPROVED** [review]
Ensure this is tested on internal platforms

**Pull request merged**